go-controller/pkg/ovn/controller/services/services_controller.go

```diff
@@ -346,19 +361,24 @@ func (c *Controller) syncService(key string) error {
 	}
 
 	// Build the abstract LB configs for this service
-	perNodeConfigs, clusterConfigs := buildServiceLBConfigs(service, endpointSlices)
+	perNodeConfigs, templateConfigs, clusterConfigs := buildServiceLBConfigs(service, endpointSlices,
+		c.useLBGroups, c.useTemplates)
 	klog.V(5).Infof("Built service %s LB cluster-wide configs %#v", key, clusterConfigs)
 	klog.V(5).Infof("Built service %s LB per-node configs %#v", key, perNodeConfigs)
+	klog.V(5).Infof("Built service %s LB template configs %#v", key, templateConfigs)
 
 	// Convert the LB configs in to load-balancer objects
 	nodeInfos := c.nodeTracker.allNodes()
 	clusterLBs := buildClusterLBs(service, clusterConfigs, nodeInfos, c.useLBGroups)
+	templateLBs := buildTemplateLBs(service, templateConfigs, nodeInfos,
```

Comment:
@dceara this is a bit of an unrelated question, but I will ask it here before I forget. If node ip changed, the template will definitely not need to, but the underlying code in ovn-controller will have to. Is there aomething else the cms needs to handle wrt to the service on that node. 
Ref: https://github.com/ovn-org/ovn-kubernetes/pull/3131

Reply:
It should be fine as is.  The node IP change should trigger an update to be received by the `nodeTracker` which will eventually call `updateNodeInfo()` which ends with:
```
	klog.Infof("Node %s switch + router changed, syncing services", nodeName)
	// Resync all services
	nt.resyncFn()
```

That's actually `RequestFullSync()` which does:
```
	// Resync node IP templates.
	if c.useTemplates {
		nodeIPTemplates := []libovsdbops.TemplateMap{
			{
				c.nodeTracker.nodeIPV4Template.Name: c.nodeTracker.nodeIPV4Template,
				c.nodeTracker.nodeIPV6Template.Name: c.nodeTracker.nodeIPV6Template,
			},
		}
		if err := libovsdbops.CreateOrUpdateChassisTemplateVar(c.nbClient, nodeIPTemplates...); err != nil {
			klog.Errorf("Could not sync node IP templates")
			return
		}
	}
```